vignettes/wrappers.Rmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "How to write a function that wraps glue"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{How to write a function that wraps glue}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(glue)
+```
+
+Imagine that you want to call `glue()` repeatedly inside your own code (e.g. in your own package) with a non-default value for one or more arguments.
+For example, maybe you anticipate producing R code where `{` and `}` have specific syntactic meaning.
+Therefore, you'd prefer to use `<<` and `>>` as the opening and closing delimiters for expressions in `glue()`.
+
+Spoiler alert: here's the correct way to write such a wrapper:
+
+```{r}
+myglue <- function(..., .envir = parent.frame()) {
```

Comment:
Can we find a more compelling example based on one of our glue wrapper? I think it's pretty unusual to want to change the delimiters.

Reply:
Based on my own experience and, now, from GitHub searches, I don't have any better ideas. People do really do write wrappers to fiddle with the delimiters and I think other examples are probably too exotic.

But if there's a concrete alternative, I'm happy to have a go.